tests/testthat/test-serializer.R

```diff
@@ -268,3 +268,75 @@ test_that("Windows encoding", {
 
   expect_type(ser$deserialize(), "list") # could fail on Windows
 })
+
+# Compressed -------------------------------------------------------------------
+
+test_that("generates correct path", {
+  aa <- Compressed$new("path", "name")
+  expect_equal(aa$path, "path/name.qs2")
+})
+
+test_that("generates expected compressed data", {
+  local_vcr_configure(json_pretty = TRUE)
+  local_mocked_bindings(
+    cur_time = function(tz) "2024-01-01 12:00:00",
+    pkg_versions = function() "<package_versions>"
+  )
+
+  request <- Request$new(method = "GET", uri = "http://example.com")
+  response <- VcrResponse$new(status = 200L, list(name = "val"), body = "body")
+  interaction <- list(request = request, response = response)
+
+  ser <- Compressed$new(withr::local_tempdir(), "serialize")
+  ser$serialize(list(interaction))
+
+  expect_snapshot(writeLines(qs2::qs_read(ser$path)))
```

Comment:
I don't think I'd bother with this snapshot tests since this is fundamentally a binary file format; there's no way to easily confirm that you're getting the correct output written to disk. (But I also don't think it's necessary because the majority of the work is now done in `encode_interactions()`.

Reply:
Good point